operations/mimir-mixin/alerts/alerts.libsonnet

```diff
@@ -550,18 +550,22 @@ local utils = import 'mixin-utils/utils.libsonnet';
             (
               # We use RSS instead of working set memory because of the ingester's extensive usage of mmap.
               # See: https://github.com/grafana/mimir/issues/2466
-              container_memory_rss{container="ingester"}
+              container_memory_rss{container=~"(%(ingester)s|%(mimir_write)s|%(mimir_backend)s)"}
                 /
-              ( container_spec_memory_limit_bytes{container="ingester"} > 0 )
+              ( container_spec_memory_limit_bytes{container=~"(%(ingester)s|%(mimir_write)s|%(mimir_backend)s)"} > 0 )
             ) > 0.65
-          |||,
+          ||| % {
+            ingester: $._config.instance_names.ingester,
```

Comment:
I think it should be `container_names` not `instance_names`.

_Comment applies everywhere else._

Reply:
So I think in the case of the `ingester`, we would be replicating the logic of `helmCompatibleMatcher`, `baremetalCompatibleMatcher`, and `instanceMatcher`, to get the same regex that we can already get in `instance_names`.

As for mimir_backend, in `container_names` it uses `componentGroups.backend` meaning that it'd create a regex for `['query_scheduler', 'ruler_query_scheduler', 'ruler', 'store_gateway', 'compactor', 'alertmanager', 'overrides_exporter', 'mimir_backend']` which would over-include services for microservice deployment mode.

Reply:
> So I think in the case of the ingester, we would be replicating the logic of helmCompatibleMatcher, baremetalCompatibleMatcher, and instanceMatcher, to get the same regex that we can already get in instance_names.

I'm not sure about it. Few things:
1. Helm: the container name shouldn't change, so should always be `ingester` (the `helmCompatibleMatcher()` is used to match the deployment/pod which can have the `mimir-` prefix)
2. Baremetal: there are no `container_*` metrics anyway there

Reply:
okay, makes sense. I had to add an extra selector for just the read-write containers but it looks neater now.